include/nn/util.h

```diff
@@ -58,3 +58,18 @@ struct BitFlagSet {};
 
 void ReferSymbol(const void*);
 }  // namespace nn
+
+// todo: move with more info
+#define NN_NO_COPY(CLASS)                                                                          \
+private:                                                                                           \
```

Comment:
now that we're using `= delete;` I think `private:` isn't really necessary anymore (and removing it lets us get rid of a gotcha)

Reply:
It's necessary, the types have those functions as private